### PR TITLE
ISPN-10842 store migrator docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-10842

Needs backport to 10.1.x

### CODE REVIEW

File: documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc [hunk 1]
```diff
@@ -1,8 +1,43 @@
 [id='cache_store_migration']
 :context: configure
-= Migrating Cache Stores Between {brandname} Versions
+= Migrating Data Between {brandname} Versions
+Upgrading to the latest {brandname} version typically involves creating new
+clusters with the target version running in parallel to the version from which
+you are migrating. You can then create a shared persistent cache store between
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc]: Should we link to our rolling upgrade docs here?

File: documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc [hunk 2]
```diff
@@ -1,8 +1,43 @@
...
+the two clusters and switch client connections to the target.
 
-include::{topics}/persistence_storemigrator.adoc[leveloffset=+1]
+However, functional differences between {brandname} versions mean that backwards
+compatibility is not always possible.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc]: It's 100% no possible between Major versions e.g. Infinispan 8 -> 9 -> 10.

File: documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc [hunk 3]
```diff
@@ -1,8 +1,43 @@
...
+endif::productized[]
+ifdef::productized[]
+8.0
+endif::productized[]
+cannot read from cache stores with previous {brandname} versions.
+
+In other cases {brandname} versions deprecate or remove cache store
+implementations, such as JDBC Mixed and Binary stores.
+
+To overcome incompatibility with cache stores, {brandname} provides a utility
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc]: incompatibilities

File: documentation/src/main/asciidoc/topics/con_store_migrator.adoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
+[id='store_migrator']
+= Cache Store Migrator
+{brandname} provides the `StoreMigrator.java` utility that reads data from
+cache stores from previous versions and rewrites it for compatibility with the
+latest {brandname} version.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/con_store_migrator.adoc]: "from a previous version's cache store and recreates it using the latest {brandname} implementation" ?

File: documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/stories/assembly_migrating_cache_stores.adoc]: tbh I'd rather reuse the content as part of an upgrading guide. essentially we'd have the same content under Configuration > Persistence and Upgrading. I'll set this up and show you. explaining here might not be clear.

File: documentation/src/main/asciidoc/stories/assembly_migrating_data.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+clusters with the target version running in parallel to the version from which
+you are migrating. You can then create a shared persistent cache store between
+the two clusters and switch client connections to the target.
+
+However, functional differences between major {brandname} versions mean that
+backwards compatibility is not possible.
+
+For example, {brandname}
+ifndef::productized[]
+10.1
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/stories/assembly_migrating_data.adoc]: The protostream conversion actually happened in 10.0

**Comment by ryanemerson** [comment]
Thanks @oraNod

**Pull request merged**